File: Plugins/Aspose_Barcode_Java_for_Ruby/lib/asposebarcodejava/BarcodeImage/generatebarcodewithemptybars.rb
module Asposebarcodejava
  module GenerateBarcodeWithEmptyBars
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate barcode object
        symbology = Rjb::import('com.aspose.barcode.Symbology')
        bb = Rjb::import('com.aspose.barcode.BarCodeBuilder').new("TEXT", symbology.Code128)

        # Set the FilledBars property to false
        bb.setFilledBars(false)

        bb.save(data_dir + "GenerateBarcodeWithEmptyBars.jpg")

        # Display Status.
        puts "Generate Barcode With Empty Bars, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/barcode/examples/barcode_image/basic_features/GenerateBarcodeWithEmptyBars.java
package com.aspose.barcode.examples.barcode_image.basic_features;

import java.io.IOException;

import com.aspose.barcode.BarCodeImageFormat;
import com.aspose.barcode.examples.Utils;
import com.aspose.barcode.generation.BarcodeGenerator;

public class GenerateBarcodeWithEmptyBars {

	public static void main(String[] args) throws IOException {
		//ExStart: GenerateBarcodeWithEmptyBars
		// The path to the resource directory.
		String dataDir = Utils.getDataDir(GenerateBarcodeWithEmptyBars.class) + "BarcodeImage/BasicFeatures/";
		
		//Create an instance of BarcodeGenerator and initialize it with CodeText and Symbology
		BarcodeGenerator generator = new BarcodeGenerator(com.aspose.barcode.EncodeTypes.CODE_128, "TEXT");
		
		//Set the FilledBars property to false
		generator.getParameters().getBarcode().setFilledBars(false);
		
		//Save the resultant barcode image on disk
		generator.save(dataDir + "barcodeWithEmptyBars.png", BarCodeImageFormat.PNG);
		//ExEnd: GenerateBarcodeWithEmptyBars
	}

}
